src/math/trig.c

```diff
@@ -32,12 +32,30 @@ const int32_t cordic_sine_cos_piovertwo_q30fl  = Q_CONVERT_FLOAT(_M_PI / 2, 30);
 const int32_t cord_sincos_piovertwo_q28fl  = Q_CONVERT_FLOAT(_M_PI / 2, 28);
 /* 843314857,  deg = 90.000000	*/
 const int32_t cord_sincos_piovertwo_q29fl  = Q_CONVERT_FLOAT(_M_PI / 2, 29);
+/* arc trignometry constant*/
+/**
+ * \+----------+----------------------------------------+--------------------+-------------------+
+ * \|values    |		Q_CONVERT_FLOAT		|(180/pi)*angleInRad |(pi/180)*angleInDeg|
+ * \+----------+----------------------------------------+--------------------+-------------------+
+ * \|379625062 | Q_CONVERT_FLOAT(1.4142135605216026, 28)|    81.0284683480568| 1.41421356052160  |
+ * \|1073741824| Q_CONVERT_FLOAT(1.0000000000000000, 30)|    57.2957795130823| 1.00000000000000  |
+ * \|843314856 | Q_CONVERT_FLOAT(1.5707963258028030, 29)|    89.9999999431572| 1.57079632580280  |
+ * \|1686629713| Q_CONVERT_FLOAT(1.5707963267341256, 30)|    89.9999999965181| 1.57079632673413  |
+ * \+----------+----------------------------------------+--------------------+-------------------+
+ */
+/* 379625062,  deg = 81.0284683480568475 or round(1.4142135605216026*2^28) */
+const int32_t cord_arcsincos_q28fl  = Q_CONVERT_FLOAT(1.4142135605216026 / 2, 28);
+/* 843314856 , deg = 89.9999999431572348 or round(pi/2*2^29) */
+/* 843314857, Bug in calculation, so using hardcoded value */
+const int32_t cord_arcsincos_piovertwo_q29fl  = Q_CONVERT_FLOAT(_M_PI / 2, 29);
+/* 1073741824, deg = 57.2957795130823229 or round(1*2^30)*/
+const int32_t cord_arcsincos_q30fl  = Q_CONVERT_FLOAT(1.0000000000000000, 30);
 
 /**
- * \CORDIC-based approximation of sine and cosine
+ * \CORDIC-based approximation of sine, cosine and complex exponential
  */
-void cordic_sin_cos(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t *b_yn, int32_t *xn,
-		    int32_t *th_cdc_fxp)
+void cordic_approx(int32_t th_rad_fxp, cordic_cfg type, int32_t *sign, int32_t *b_yn, int32_t *xn,
+		   int32_t *th_cdc_fxp)
```

Comment:
might be better to have a struct to pass back so you have to worry about less args

Reply:
Thanks !
since not all elements are return values ,so avoiding structure as args in function call